python/vision_explanation_methods/explanations/drise.py

```diff
@@ -257,3 +260,118 @@ def DRISE_saliency(
+        image_tensor: pd.DataFrame,
+        target_detections: List[DetectionRecord],
+        number_of_masks: int,
+        mask_res: Tuple[int, int] = (16, 16),
+        mask_padding: Optional[int] = None,
+        device: str = "cpu",
+        verbose: bool = False,
+) -> List[torch.Tensor]:
+    """Compute DRISE saliency map
+
+    :param model: Object detection model wrapped for occlusion
+    :type model: OcclusionModelWrapper
+    :param target_detections: Baseline detections to get saliency
+        maps for
+    :type target_detections: List of Detection Records
+    :param number_of_masks: Number of masks to use for saliency
+    :type number_of_masks: int
+    :param mask_res: Resolution of mask before scale up
+    :type maks_res: Tuple of ints
+    :param mask_padding: How much to pad the mask before cropping
+    :type: Optional int
+    :device: Device to use to run the function
+    :type: str
+    :return: A list of tensors, one tensor for each image. Each tensor
+        is of shape [D, 3, W, H], and [i ,3 W, H] is the saliency map
+        associated with detection i.
+    :rtype: List torch.Tensor
+    """
+
+    if not isinstance(image_tensor, pd.DataFrame):
+        raise ValueError(
+            "TypeError: Image needs to be a torch.Tensor or pd.DataFrame")
+    if not image_tensor.shape[0] == 1:
+        raise ValueError(
+            "Currently only one image supported for AutoML mlflow model")
+
+    img_size = image_tensor.loc[0, 'image_size']
+
+    if mask_padding is None:
+        mask_padding = int(max(
+            img_size[0] / mask_res[0], img_size[1] / mask_res[1]))
+
+    mask_records = []
+
+    mask_iterator = tqdm.tqdm(range(number_of_masks)) if verbose \
+        else range(number_of_masks)
+
+    for _ in mask_iterator:
+        # Converts image base64 to a tensor
+        # Fuses mask tensor with image tensor
+        # Converts fused image tensor to base64
+        mask = generate_mask(mask_res, img_size, mask_padding)
+
+        # Currently only supports single image
+        img_tens = convert_base64_to_tensor(
+            image_tensor.loc[0, 'image'])
+
+        masked_image = fuse_mask(img_tens.to(device), mask.to(device))
+
+        masked_image_str, masked_image_size = convert_tensor_to_base64(
+            masked_image)
+
+        masked_df = pd.DataFrame(
+            data=[[masked_image_str, masked_image_size]],
+            columns=['image', "image_size"],
+        )
+
+        masked_detections = model.predict(masked_df)
+
+        affinity_scores = []
+        for (target_detection, masked_detection) in zip(target_detections,
+                                                        masked_detections):
+            affinity_scores.append(
+                compute_affinity_scores(target_detection, masked_detection))
+
+        mask_records.append(MaskAffinityRecord(
+            mask=mask.to("cpu"),
+            affinity_scores=[s.detach().to("cpu") for s in affinity_scores])
```

Comment:
```suggestion
            affinity_scores=[s.detach().cpu() for s in affinity_scores])
```
Nit: avoid hardcoded str value.

Reply:
fixed